rfc/0009-qsl-structure.md

```diff
@@ -0,0 +1,133 @@
+### Libraries
+Each library would be its own folder in the QSL monorepo, with a dedicated team responsible for triage, review, and accepting/denying pull requests. In contrast to Fabric or Forge's system of only one person signing off for the entire standard library, no authority higher than the people on the team would be required to sign off on the PR before it is merged. More clarification on team structure is expected to come in a future RFC.
+
+Libraries only exist for internal structure--there is no fat jar for, say, all `worldgen` modules, neither is there a fat jar for all of QSL. This also means that QSL and its libraries have no versioning.
+### Modules
+Modules follow the following general rules:
+- Maven group of `org.quiltmc.qsl.$library`
+- Artifact id of `mod-id`
+- Semantic Versioning. Hashes are *not* used to identify modules. 
+- Can only depend on modules from `core` or within their own library.
+
+To modders, most mods would not explicitly depend on libraries or their modules, but Loom would automatically determine the modules to depend on by scanning at compile time, and those modules will be downloaded at runtime by Loader.
+### Top-level Structure
+Keep in mind that the modules listed here are the names of the current Fabric API modules, and the final internal structure of QSL **will not** be the same as Fabric API.
+- `core` - All modules here are implicitly depended on by mods by default, even if they do not use any code from them. Mods can choose to opt-out from this, but if they choose to use the fat QSL dependency they are responsible for ensuring these modules don't get transitively pulled in.
+    - Lifecycle Events
+    	- Here because of the Networking API
+    	- Some parts might be split into `entities` and `world`
+    - Networking API (maybe with relevant parts split into other modules)
+        - Registry Sync (merged in)
+        - (future) Proper Handshake API
+    - Crash Reports
+    - Resource Loader
+- `transfer`
+    - API Lookup API
+    - fluid api
+    - item transfer api
+- `command` // This really is anything server-side only related to administration, but "command" is a nice word to describe the primary use-case.
+    - Command API
+    - Game Rule API
+    - (future) Permissions API
+- `rendering` // This entire library should be client-side only
+	- BlockRenderLayer Registration
+	- Indigo
+	    - need an easy way to exclude this
+	- Models
+	- Renderer API
+		- This may need a better name in a spot?
+	- Rendering
+	- Renderer Registries
+	- Rendering Fluids
+	- Textures
+	- Object Builder API (FabricModelPredicateProviderRegistry)
+- `data` - Tools for working with, generating, or consuming data (e.g. datapacks, advancements, recipes, tags)
+    - (future) Recipe API
+    - (future) Datagen APIs
+    - Rendering Data Attachment
```

Comment:
Rendering data attachment is not exactly related to generation or modification or consumption of data.